__tests__/app.test.js

```diff
@@ -533,4 +533,49 @@ describe("DELETE /api/comments/:comment_id", () => {
   });
 });
 
-
+describe("GET /api/users", () => {
+  test("response code is 200", () => {
+    return request(app).get("/api/users").expect(200);
+  });
+  test("users table exists in database", () => {
+    return db
+      .query(
+        `SELECT EXISTS (
+                          SELECT FROM 
+                          information_schema.tables 
+                          WHERE 
+                          table_name = 'users'
+                          );`
+      )
+      .then(({ rows: [{ exists }] }) => {
+        expect(exists).toBe(true);
+      });
+  });
+  test("returns an array", () => {
+    return request(app)
+      .get("/api/users")
+      .then((res) => {
+        expect(Array.isArray(res.body.users)).toBe(true);
+      });
+  });
+  test("Returned array contains correct number of objects", () => {
+    return request(app)
+      .get("/api/users")
+      .expect(200)
+      .then((response) => {
+        const result = response.body.users;
+        expect(result.length).toBe(4);
+      });
+  });
+  test("returned user object has correct keys", () => {
+    return request(app)
+      .get("/api/users")
+      .expect(200)
+      .then((response) => {
+        const ActualkeysArray = Object.keys(response.body.users[0]);
+        expect(ActualkeysArray.includes("username")).toEqual(true);
+        expect(ActualkeysArray.includes("name")).toEqual(true);
+        expect(ActualkeysArray.includes("avatar_url")).toEqual(true);
```

Comment:
This technically works, but there's more succinct and readable testing methods we can use. Have a go at using .toMatchObject instead when trying to test the properties of objects. It was built for this exact purpose and gives us some nice syntax, allowing us to also test the value and type of the properties! Have a look at this possible refactor:

```
.then((response) => {
         expect(response.body.users[0]).toMatchObject({
         // Now you define the properties you expect there to be inside here
         username: expect.any(String)
         // we can use expect.any and a datatype if you don't know what the property will exactly be
         // but keep in mind you can also put a direct value in place here e.g.
         name: 'Modou'
        // ... other properties

});
```

Have a go with this testing pattern for objects and see if it makes things more readable and easier to work with (not to mention writing stronger tests too!)